source/common/router/router.cc

```diff
@@ -656,7 +660,15 @@ void Filter::onUpstreamReset(Http::StreamResetReason reset_reason,
 
   const bool dropped = reset_reason == Http::StreamResetReason::Overflow;
   callbacks_->streamInfo().setUpstreamTransportFailureReason(transport_failure_reason);
-  onUpstreamAbort(Http::Code::ServiceUnavailable, response_flags, body, dropped);
+  // TODO(reviewer) do we want structure here rather than string?
+  // I think in practice if there's a transport failure it takes precedence,
+  // otherwise there was an application error. So maybe just converting both
+  // transport_failure_reasons and reset_reasons to the snake_case is
+  // sufficient?  But maybe there's cases where we want multiple levels to add
+  // their information.
+  std::string details =
+      absl::StrCat("upstream_reset{", reset_reason, ",", transport_failure_reason, "}");
```

Comment:
Both the reset reason and the transport failure reason are already available in the StreamInfo so why include them also in the rc details string? Seems simpler to just have the rc details by a fixed string like "upstream reset" which can be easily checked for and then access logs can look at the other fields if they want more specifics.

Reply:
That's totally doable.  What I don't like about that is I feel like it encourages a paradigm of "look elsewhere for the underlying failure cause"

So today we have have "rc details" which might point us to "upstream failure" which would require us to know which of transport failure and/or reset reason to debug.  But then we could have a generic "cache failure" which could tell you to look at the cache details (was the entry not present?  expired?  etc) and "WASM failure" with a bunch of reasons the code failed to execute. 

I want all of the information to be communicated in access logs and headers and I'd mildly prefer to combine the data in the string as it's created rather than to encourage each complicated filter to split out layers of reasons.

Reply:
That makes sense. It does make it a bit difficult to programmatically check if the response code details from a StreamInfo matches this "constant" but perhaps that is not a big deal - we could add some kind of regex matching if we really want that.